### PR TITLE
code-push logout --local

### PR DESCRIPTION
**Pull request by geof90**
Currently, logout always deletes the access key on the server. Calling it with the additional --local flag removes the access key saved locally, but does not delete the access key on the server so that the same key can be used for future logins.

Bumped the CLI version to 1.1.2-beta, and also updated docs.

### CODE REVIEW

File: cli/README.md [hunk 1]
```diff
@@ -54,6 +54,12 @@ After creating the new key, you can specify its value using the `--accessKey` fl
 code-push login --accessKey <accessKey>
 ```
 
+After using the key, if you want to log out of your current session, but still want to be able to re-use the same key for future logins, run the following command:
```
Comment (dtivel) [inline cli/README.md]: "Reuse" is not hyphenated.

**Comment by dtivel** [comment]
looks good

File: cli/README.md [hunk 2]
```diff
@@ -54,6 +54,12 @@ After creating the new key, you can specify its value using the `--accessKey` fl
 code-push login --accessKey <accessKey>
 ```
 
+After using the key, if you want to log out of your current session, but still want to be able to reuse the same key for future logins, run the following command:
```
Comment (silhouettes) [inline cli/README.md]: How about:

If you want to log out of your current session without invalidating the key for future sessions, run the following command:

File: cli/script/command-parser.ts [hunk 3]
```diff
@@ -212,6 +212,12 @@ var argv = yargs.usage(USAGE_PREFIX + " <command>")
     .command("logout", "Log out of the current session", (yargs: yargs.Argv) => {
         isValidCommandCategory = true;
         isValidCommand = true;
+        yargs.usage(USAGE_PREFIX + " logout [--local <true|false>]")
```
Comment (silhouettes) [inline cli/script/command-parser.ts]: Can we not specify <true|false> here? The convention for these flags is that if they exist, they change something, and if they don't, it's the default behavior

File: cli/script/command-parser.ts [hunk 4]
```diff
@@ -212,6 +212,12 @@ var argv = yargs.usage(USAGE_PREFIX + " <command>")
...
+            .demand(/*count*/ 1, /*max*/ 1)  // Require exactly one non-option argument.
+            .example("logout", "Log out and also remove the access key used for the current session.")
```
Comment (silhouettes) [inline cli/script/command-parser.ts]: Could we try:
Log out and also invalidate the access key for future sessions

Comment (silhouettes) [inline cli/script/command-parser.ts]: Actually this is fine, forget I said anything!

File: cli/script/command-parser.ts [hunk 5]
```diff
@@ -212,6 +212,12 @@ var argv = yargs.usage(USAGE_PREFIX + " <command>")
...
+            .example("logout --local", "Log out but still allow the use of the same access key for future logins.")
```
Comment (silhouettes) [inline cli/script/command-parser.ts]: Can we remove the word 'still' here?

File: cli/script/command-parser.ts (hunk 3 above)
Comment (geof90) [inline cli/script/command-parser.ts]: Should I also remove it for --mandatory and --verbose in other commands?

Comment (silhouettes) [inline cli/script/command-parser.ts]: Ah I see, you were matching what was there before. I say yes for sure, the convention in CLI's (e.g. git) is definitely that you don't need to specify 'true' or 'false'.

If you don't mind, could you also update the README?

**Comment by silhouettes** [comment]
Nice work, looks good to me! Sorry for being nitpicky.

**Pull request merged**